build.sbt

```
@@ -104,11 +103,33 @@ lazy val sbtPlugin = project
   )
   .dependsOn(core)
 
+def isScala2(v: Option[(Long, Long)]): Boolean = v.exists(_._1 == 2)
+def isScala3(v: Option[(Long, Long)]): Boolean = v.exists(_._1 == 3)
+
+def multiScalaDirectories(root: File, scalaVersion: String) = {
+  val base = root / "src" / "main"
+  val result = mutable.ListBuffer.empty[File]
+  val partialVersion = CrossVersion.partialVersion(scalaVersion)
+  partialVersion.collect { case (major, minor) =>
+    result += base / s"scala-$major.$minor"
+  }
+  if (isScala2(partialVersion)) {
+    result += base / "scala-2"
+  }
+  if (isScala3(partialVersion)) {
+    result += base / "scala-3"
+  }
+  result += base / s"scala-$scalaVersion"
+  result.toList
+}
+
 lazy val expressionCompiler = project
   .in(file("expression-compiler"))
   .settings(
     name := "scala-expression-compiler",
     crossScalaVersions := Seq(
+      "3.1.1-RC1",
```

```suggestion
      "3.1.1",
```
we can use release version